TrafficCapture/trafficReplayer/src/test/java/org/opensearch/migrations/replay/datahandlers/http/HttpJsonTransformingConsumerTest.java

```diff
@@ -96,25 +100,30 @@ public void testRequestProcessing(Integer attemptedChunks, Boolean hostTransform
         }
 
         var chunks = Math.min(attemptedChunks, testBytes.length);
-        sliceRandomChunks(testBytes, chunks).forEach(chunk -> transformingHandler.consumeBytes(chunk));
+        sliceRandomChunks(testBytes, chunks).forEach(transformingHandler::consumeBytes);
 
         var returnedResponse = transformingHandler.finalizeRequest().get();
 
         var expectedBytes = (hostTransformation)
-            ? new String(testBytes, StandardCharsets.UTF_8).replace("foo.example", "bar.example")
-                .getBytes(StandardCharsets.UTF_8)
+            ? replaceBytes(testBytes,
```

Comment:
This enables us to test with gzipped requests without messing up the body bytes by encoding in utf8